pkgs/development/ruby-modules/with-packages/default.nix

```diff
@@ -0,0 +1,59 @@
+{ stdenv, lib, buildEnv, buildRubyGem, ruby, gemConfig, makeWrapper }:
+
+/*
+Example usage:
+nix-shell -E "(import <nixpkgs> {}).ruby.withPackages (pkgs: with pkgs; [ pry nokogiri ])"
+
+You can also use this for writing ruby scripts that run anywhere that has nix
+using a nix-shell shebang:
+  #!/usr/bin/env nix-shell
+  #!nix-shell -i ruby -p "ruby.withPackages (pkgs: with pkgs; [ pry nokogiri ])"
+
+
+Run the following in the nixpkgs root directory to update the ruby-packages.nix:
+./maintainers/scripts/update-ruby-packages
+
+To request addition of a gem, just put it into the Gemfile in this directory,
+update ruby-packages.nix, and make a PR mentioning @manveru.
+
+Please note that this will only provide you with libraries you can require
+within ruby or irb. The executables of gems aren't added to your PATH.
+For executables, we recommend
+*/
+
+let
+  functions = import ../bundled-common/functions.nix { inherit lib gemConfig; };
+
+  gems = lib.mapAttrs (name: initialAttrs:
+    let
+      attrs = functions.applyGemConfigs ({ inherit ruby; gemName = name; } // initialAttrs);
+    in
+      buildRubyGem (functions.composeGemAttrs ruby gems name attrs)
+  ) (import ../../../top-level/ruby-packages.nix);
+
+  withPackages = selector:
+    let
+      selected = selector gems;
+      gemEnv = buildEnv { name = "ruby-gems"; paths = selected; };
+      wrappedRuby = stdenv.mkDerivation {
+        name = "wrapped-${ruby.name}";
+        nativeBuildInputs = [ makeWrapper ];
+        buildCommand = ''
+          mkdir -p $out/bin
+          for i in ${ruby}/bin/*; do
+            makeWrapper "$i" $out/bin/$(basename "$i") \
+              --set GEM_HOME ${gemEnv}/${ruby.gemPath} \
+              --set GEM_PATH ${gemEnv}/${ruby.gemPath}
+          done
+        '';
+      };
+    in buildEnv {
+      name = "${ruby.name}-with-packages";
+      paths = [ wrappedRuby ] ++ selected;
```

Comment:
Will this mask other paths in Ruby that might be desirable if it were included, for example, in `environment.systemPackages`?

Reply:
It will most likely cause conflicts, since both are at default priority. So if you put both `ruby` and `ruby.withPackages` in there, it will conflict. I don't think that's a problem, because otherwise they're not usable anyway if the interpreter can't find them.